tests/diff/float_values_test.go

```diff
@@ -61,6 +63,87 @@ func TestFloatValues(t *testing.T) {
 
 				_, err := db.Collection("insert-"+name).InsertOne(ctx, tc.doc)
 
+				t.Run("FerretDB", func(t *testing.T) {
+					assertEqualError(t, tc.expected, err)
+				})
+
+				t.Run("MongoDB", func(t *testing.T) {
+					require.NoError(t, err)
+				})
+			})
+		}
+	})
+
+	t.Run("Update", func(t *testing.T) {
+		t.Parallel()
+
+		// to set UpdateOptions.Upsert.
+		boolPointer := func(b bool) *bool {
+			return &b
+		}
+
+		for name, tc := range map[string]struct {
+			filter   bson.D
+			update   bson.D
+			opts     options.UpdateOptions
+			expected mongo.CommandError
+		}{
+			"NaN": {
+				filter: bson.D{{"_id", "1"}},
```

Comment:
I'm a bit skeptical about those filters because they won't find anything (there is currently no data in this collection the DB). Can we successfully insert some data, and then call `update`?